shared/actions/chat.js

```diff
@@ -43,17 +43,22 @@ function _inboxToConversations (inbox: GetInboxAndUnboxLocalRes): List<InboxStat
       return null
     }
 
-    const recentMessage: ?MessageUnboxed = (convo.maxMessages || []).find(message => (
-      message.state === LocalMessageUnboxedState.valid &&
-      message.valid &&
-      message.valid.messageBody.messageType === CommonMessageType.text
-    ))
-
     let snippet
-    try {
-      // $FlowIssue doens't understand try
-      snippet = makeSnippet(recentMessage.valid.messageBody.text.body, 100)
-    } catch (_) { }
+    (convo.maxMessages || []).some(message => {
+      if (message.state === LocalMessageUnboxedState.valid && message.valid) {
+        switch (message.valid.messageBody.messageType) {
+          case CommonMessageType.text:
+            snippet = makeSnippet(message.valid.messageBody.text && message.valid.messageBody.text.body, 100)
```

Comment:
maybe, lets wait till we see that's the case. we only use snippets in the inbox so far